packages/web-components/custom-elements-manifest.plugins.js

```diff
@@ -0,0 +1,167 @@
+import { readFileSync } from "fs";
+import path from "path";
+import ts from "typescript";
+import { fileURLToPath } from 'url';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+const getTagNameFromCommentInDefinitionFile = function (definitionPathName) {
+  const indexFilePath = path.resolve(__dirname, `./${definitionPathName}`);
+  let name;
+
+  try {
+    let sourceFile = ts.createSourceFile(
+      indexFilePath,
+      readFileSync(indexFilePath).toString(),
+      ts.ScriptTarget.ES2015,
+      /*setParentNodes */ true
+    );
+
+    if (Array.isArray(sourceFile.statements)) {
+      sourceFile.statements.forEach((statement) => {
+        if (Array.isArray(statement.jsDoc) && statement.jsDoc[0].tags !== undefined) {
+          statement.jsDoc.forEach((jsDoc) => {
+            if (Array.isArray(jsDoc.tags)) {
+              jsDoc.tags.forEach((tag) => {
+                if (typeof tag.comment === "string" && tag.comment.startsWith("HTML Element:")) {
+                  name = tag.comment.match(/<(.*)>/)[1].replace("\\", "");
+                }
+              });
+            }
+          })
+        }
+      })
+    }
```

Comment:
Nit: to improve readability, maybe test negative cases and return? Something like:

```js
if (!Array.isArray(...)) {
  return;
}
sourceFile.statements.forEach(...);
```

Reply:
Agree with @marchbox's suggestion these nested ifs could be easier to read. Can `sourceFile.statements`, `statement.jsDoc`, or `jsDoc.tags` _not_ be arrays?